### DISCUSSION TITLE
AWS Amplify robot.txt errors lighthouse audit

### DISCUSSION THREAD

**Discussion by 3lonious** [vercel/vercel#4674] [Help] [answered]
i cant seem to get these errors to go away, lighthouse audit is telling me my site will be indexed improperly. ive tried everything and cant seem to figure this out. im deploying via github and AWS amplify
<img width="535" alt="Screen Shot 2020-06-17 at 11 07 24 AM" src="https://user-images.githubusercontent.com/46793692/84933670-19527b00-b08b-11ea-9cf6-9e3d25c65038.png">
<img width="533" alt="Screen Shot 2020-06-17 at 11 07 44 AM" src="https://user-images.githubusercontent.com/46793692/84933685-1f485c00-b08b-11ea-97e9-8f451044a553.png">

**Comment by styfle** [comment]
Which framework are you using?

Make sure you have a [`robots.txt`](https://www.robotstxt.org/) file in your [Output Directory](
https://vercel.com/docs/v2/build-step#output-directory).

```
User-agent: *
Disallow:
```

**Comment by 3lonious** [comment]
so i figured out how to make sure the robot.txt ends up in the right build/output folder but im still getting these errors, its basicly seeing the entire app code as the robot.txt file?
<img width="535" alt="Screen Shot 2020-06-19 at 7 58 00 AM" src="https://user-images.githubusercontent.com/46793692/85146631-a1a75c00-b202-11ea-8ef5-db3f7f4f03c2.png">

**Comment by styfle** [comment]
Do you see the correct file when visiting `/robots.txt`?

For example `https://example.com/robots.txt`?

**Comment by 3lonious** [comment]
yes i do, its there :)

<img width="200" alt="Screen Shot 2020-06-19 at 8 00 27 AM" src="https://user-images.githubusercontent.com/46793692/85146948-ffd43f00-b202-11ea-98b0-69f0cdb13774.png">

**Comment by styfle** [comment]
Great! So it sounds like the response was cached.

You can try clearing your browser cache if you're running lighthouse locally.

If not, deploy the new file to clear the CDN cache just in case.

https://vercel.com/docs/v2/edge-network/caching

**Comment by 3lonious** [comment]
im deploying thru AWS amplify

**Comment by styfle** [comment]
Perhaps you should ask the AWS Amplify why it isn't working.

Is there a reason why you chose AWS Amplify instead of [Vercel.com](http://vercel.com) to host your project?

**Comment by 3lonious** [comment]
i have a complicated dns route 53 set up and i found trying to move my dns over to vercel/zeit last year didnt work very well and could not replicate my dns records, i have lots of infrastructure in aws so its the best use for me. also im not sure how much my SEO gets effected by changing my dns away from AWS west region

**Comment by styfle** [comment]
> i found trying to move my dns over to vercel/zeit last year didnt work

This has been fixed recently, checkout [this tweet](https://twitter.com/matheusfrndes/status/1270376292920197122) from @matheuss.

We are still updating documentation but the idea is that you use:

- `CNAME cname.vercel-dns.com` for subdomains
- `A 76.76.21.21` for apex domains

Previously you had to use `ANAME` for apex domains which Route 53 didn't support.

**Comment by 3lonious** [comment]
oh really? so i can just grab the IP address and use it in route 53 ?

**Comment by styfle** [comment, marked as answer]
Yep, that's it! 👍 

Then you can manage DNS via Route 53 and deploy to Vercel 🚀 

You'll see a new message pop up when you add the domain to your Vercel Project.

<img src="https://user-images.githubusercontent.com/229881/85163026-396f6f00-b230-11ea-91bb-6f908660f197.png" width="500" />

**Official Announcement**: https://vercel.com/blog/new-edge-dev-infrastructure

**Comment by 3lonious** [comment]
this is the best news all week thank you

**Comment by 3lonious** [comment]
hey sorry to keep bugging you but now that im deploying on vercel im getting this error?

<img width="529" alt="Screen Shot 2020-06-19 at 10 50 32 AM" src="https://user-images.githubusercontent.com/46793692/85165490-c6a7c900-b21a-11ea-9ddb-3d24e675f8da.png">

**Comment by 3lonious** [comment]
i may have solved the issue, its a script tag maybe in the wrong place

**Comment by 3lonious** [comment]
ya so im having trouble adding my zoho chat script
  <script dangerouslySetInnerHTML={{

                                __html: `
                    if (typeof window !== 'undefined') { var $zoho=$zoho || {};$zoho.salesiq = $zoho.salesiq || {widgetcode:"CODE_HERE", values:{},ready:function(){}};var d=document;s=d.createElement("script");s.type="text/javascript";s.id="zsiqscript";s.defer=true;s.src="https://salesiq.zoho.com/widget";t=d.getElementsByTagName("script")[0];t.parentNode.insertBefore(s,t);d.write("<div id='zsiqwidget'></div>");
                    //initialise here
                    }
                    `}} />

i get this error
<img width="529" alt="Screen Shot 2020-06-19 at 10 50 32 AM" src="https://user-images.githubusercontent.com/46793692/85167774-2eabde80-b21e-11ea-8a52-ae8a94d8f4ac.png">

**Comment by styfle** [comment]
Your code snippet is calling `document.write()`.
If you remove it then it won't be invoked and lighthouse won't complain 😄 

```js
var d=document;
d.write(""); //initialise here
```

Seems like Zoho is not following good practices here.

**Comment by theodore-s-beers** [comment]
I just encountered the same problem. Thanks for the answer! It still strikes me as odd that, in the absence of a `robots.txt` file, all of `index.html` will be interpreted as `robots.txt`.
